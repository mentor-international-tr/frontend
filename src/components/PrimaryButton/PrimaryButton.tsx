import React from "react";
import PropTypes from "prop-types";
import classNames from "classnames";
import { withStyles } from "@material-ui/core/styles";
import Button from "@material-ui/core/Button";
import themeStyles from "../../lib/styles/variables";
// We can inject some CSS into the DOM.
const styles = {
  root: {
    background: themeStyles.blue,
    color: themeStyles.white,
    "&:hover": {
      background: themeStyles.darker_blue,
    },
  },
};

const PrimaryButton: any = (props: any) => {
  const { classes, children, className, ...other } = props;

  return (
    <Button
      className={classNames(classes.root, className)}
      {...other}
      data-testid="primary-button"
    >
      {children}
    </Button>
  );
};

PrimaryButton.propTypes = {
  children: PropTypes.node,
  classes: PropTypes.object.isRequired,
  className: PropTypes.string,
};

export default withStyles(styles)(PrimaryButton);
