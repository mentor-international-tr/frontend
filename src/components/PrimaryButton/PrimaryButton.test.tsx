import React from "react";
import { render, cleanup } from "react-testing-library";
import PrimaryButton from "./PrimaryButton";
import "jest-dom/extend-expect";

afterEach(cleanup);

describe("testing <PrimaryButton /> component", () => {
  it("creates snapshot", () => {
    const { asFragment } = render(
      <PrimaryButton>I am text</PrimaryButton>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it("inserts text as a child of the button", () => {
    const { getByTestId } = render(
      <PrimaryButton>test</PrimaryButton>,
    );
    expect(getByTestId("primary-button")).toHaveTextContent(
      "test",
    );
  });
});
