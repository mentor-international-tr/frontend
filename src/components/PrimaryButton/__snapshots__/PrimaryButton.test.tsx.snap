// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing <PrimaryButton /> component creates snapshot 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root-28 MuiButton-root-2 MuiButton-text-4 MuiButton-flat-7 PrimaryButton-root-1"
    data-testid="primary-button"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label-3"
    >
      I am text
    </span>
    <span
      class="MuiTouchRipple-root-31"
    />
  </button>
</DocumentFragment>
`;
